### PR TITLE
Add append method for nn.Sequential

### PR DESCRIPTION
**Pull request by jaketae**
Partially addresses https://github.com/pytorch/pytorch/issues/71249, and potentially supersedes https://github.com/pytorch/pytorch/pull/20274.

### CODE REVIEW

**Comment by jaketae** [comment]
Note that the linked PR implements more methods, whereas the current PR is more contained and only implements `append`. If need be, I can open a separate issue/PR that discusses the possibility of implementing the rest (extend, insert, etc).

**Comment by jaketae** [comment]
Hello @jbschlosser, hope it's okay to ping you here! Do you see any additional actionable item(s) to complete this PR? Thanks in advance.

File: test/test_nn.py [hunk 1]
```diff
@@ -1507,6 +1507,16 @@ def test_Sequential_delitem(self):
...
 
+    def test_Sequential_append(self):
+        l1 = nn.Linear(10, 20)
+        l2 = nn.Linear(20, 30)
+        l3 = nn.Linear(30, 40)
+        l4 = nn.Linear(40, 50)
+        n = nn.Sequential(l1, l2, l3)
+        n2 = n.append(l4)
+        self.assertEqual(n, nn.Sequential(l1, l2, l3, l4))
+        self.assertEqual(n2, nn.Sequential(l1, l2, l3, l4))
```
Comment (jbschlosser) [inline test/test_nn.py:1518]: nit: maybe also test chaining via the returned self? e.g. `n.append(...).append(...)` Not super important but would be nice

**Review by jbschlosser — COMMENTED** [review]
Hey @jaketae, thanks for the ping! This PR seemed to have slipped from my review queue somehow.

I agree with your design choices:
* `append()` should only accept a single module, following the API of Python's list. I'd expect `extend()` to be the API for extending with an iterable of modules.
* I like the idea of returning `self` to match `ModuleList`.
* I'm good with keeping this PR to `append()` and addressing the rest in a possible future PR.

Thanks for making the update!

Do you mind rebasing? Seems like some pre-existing failures broke tests and rebasing *should* fix this.

File: test/test_nn.py (hunk 1 above)
Comment (jaketae) [reply test/test_nn.py:1532]: Agreed, I added a line below that checks for chaining!

**Comment by jaketae** [comment]
Hello @jbschlosser, thank you for the review! I've rebased with master and added the chained test you mentioned in the comment. 

I've compiled a list of other potential ops we might want added to `nn` containers in a separate [RFC](https://github.com/pytorch/pytorch/issues/71329). In the meantime, I'll open a new PR for `extend`, which seems like a reasonable extension (no pun intended) to `append`.

Thank you!

**Review by jbschlosser — APPROVED** [review]
Awesome thanks for the update! LGTM

**Pull request closed without merging**